common/src/main/java/io/github/notenoughupdates/moulconfig/gui/component/SliderComponent.kt

```diff
@@ -55,25 +59,76 @@ open class SliderComponent(
         }
         mc.bindTexture(GuiTextures.SLIDER_BUTTON)
         context.renderContext.drawTexturedRect((sliderPosition - 4).toFloat(), 0F, 8F, context.height.toFloat())
+
+        context.renderContext.translate(60f, -5f, 0f)
+        componentNumberInput.render(context.translated(60, -5, componentNumberInput.width, 18))
     }
 
     fun setValueFromContext(context: GuiImmediateContext) {
-        var v: Float = context.mouseX * (maxValue - minValue) / context.width + minValue
+        var v: Float = (context.mouseX + width/3) * (maxValue - minValue) / context.width + minValue
         v = min(v.toDouble(), maxValue.toDouble()).toFloat()
         v = max(v.toDouble(), minValue.toDouble()).toFloat()
         v = Math.round(v / minStep) * minStep
         value.set(v)
     }
 
+    // I made this because I couldn't get isHovered to work with the translation
+    private fun GuiImmediateContext.isHovered(): Boolean {
+        return mouseX in 0 - width/3 until width && mouseY in 0 until height
+    }
+
     override fun mouseEvent(mouseEvent: MouseEvent, context: GuiImmediateContext): Boolean {
         if (!context.renderContext.isMouseButtonDown(0)) clicked = false
-        if (context.isHovered && mouseEvent is MouseEvent.Click && mouseEvent.mouseState && mouseEvent.mouseButton == 0) {
+        if (context.isHovered() && mouseEvent is MouseEvent.Click && mouseEvent.mouseState && mouseEvent.mouseButton == 0) {
             clicked = true
         }
         if (clicked) {
             setValueFromContext(context)
-            return true
         }
-        return false
+        return componentNumberInput.mouseEvent(mouseEvent, context.translated(45, -5, componentNumberInput.width, 18))
+    }
+
+    override fun keyboardEvent(event: KeyboardEvent, context: GuiImmediateContext): Boolean {
+        return componentNumberInput.keyboardEvent(event, context)
+    }
+
+    private val componentNumberInput by lazy {
```

Comment:
This code does not belong in the SliderComponent. The SliderComponent should just render the slider. The combination with the text component should happen outside of this class.